### PR TITLE
Cast `args` and `kwargs` to 64-bit floating point precision.

### PR DESCRIPTION
**Pull request by AlbertMitjans**
**Context:**
Jax uses `float32` by default, which may cause rounding errors.

**Description of the Change:**
In this PR we change `jax` global config to always use `float64`.

**Related GitHub Issues:**
Fixes https://github.com/PennyLaneAI/pennylane/issues/3785

### CODE REVIEW

File: pennylane/qnode.py [hunk 1]
```diff
@@ -824,6 +824,15 @@ def __call__(self, *args, **kwargs):  # pylint: disable=too-many-branches, too-m
                     "The device was switched during the call of the QNode, to avoid this behaviour define"
                     "an interface argument instead of auto."
                 )
+        if self.interface == "jax":
+            import jax  # pylint: disable=import-outside-toplevel
+            import jax.numpy as jnp  # pylint: disable=import-outside-toplevel
+
+            if jax.config.read("jax_enable_x64") is False:
+                # We cast to 64-bit floating points to avoid rounding errors
+                jax.config.update("jax_enable_x64", True)
```
Comment (AlbertMitjans) [inline pennylane/qnode.py:833]: I'm not sure if we sould change this global variable.

Comment (AlbertMitjans) [reply pennylane/qnode.py:833]: Some users might want to keep this value to `False`...

Comment (PhilipVinc) [reply pennylane/qnode.py:833]: you can set it with a context manager IIRC

Comment (PhilipVinc) [reply pennylane/qnode.py:833]: https://github.com/google/jax/blob/c657449528085f140fef8dcb6dc41f8b136ba0f8/tests/x64_context_test.py#L40

Comment (AlbertMitjans) [reply pennylane/qnode.py]: Now the config is updated during PL import.

**Comment by trbromley** [comment]
Closing this PR - we will take some time to think about the long term plan here. Thanks @AlbertMitjans!

**Pull request closed without merging**